### DISCUSSION TITLE
StaggeredLayout

### DISCUSSION THREAD

**Discussion by ShariatPanah** [dotnet/maui#7760] [Ideas]
in some cases you need to display some items that have different heights from each other (not necessarily different height, they might have the same height), so this is where the StaggeredLayout makes your life lot easier but Xamarin and .Net Maui are really missing this great and beautiful layout, although there was tons of feature request for this control to be created in Xamarin, but without any luck (lets not talk about custom renderers and blah blah blah)

so please consider implementing this layout in .Net Maui, lets show some images of what I'm talking about:

![2](https://user-images.githubusercontent.com/32509267/172026235-c7b19ef8-a4e6-41ff-9598-6cc9ba5eb316.png)

![4](https://user-images.githubusercontent.com/32509267/172026469-358a40bc-007d-4fd0-9b19-904cc7cf5379.png)

**Comment by pictos** [comment]
@ShariatPanah you can do it with FlexLayout, here's a [sample](https://github.com/kphillpotts/ArtAuction)

**Reply by ShariatPanah** [reply]
@pictos thank you Jesus, i have seen it before, it uses Column Direction of flexlayout, should I mention that for a real world application it's not a good practice to use flex layout with Column direction?

Because in most apps you have an endless vertical collection but when you use a Columned Flexlayout, you have a horizontal collection
That's not a real StaggeredLayout, is it?

I hope I made my point

**Reply by pictos** [reply]
Well if you can't use the Column direction in real-world programming we should open a bug issue on the repo, with a repro asking the team to fix it. Because everything in this framework (and the Xamarin.Forms) is for real-world programming.

**Reply by ShariatPanah** [reply]
Sir, I think you didn't get my point! I meant when you use a Columned FlexLayout, there are some problems:
1. you should specify explicit Height for FlexLayout, therefore when your items number grows you have some issues: you would have both Vertical and Horizontal Scrolling at the same time, it wouldn't be a good looking UI/UX really.

2. the items would display in Column Order, meaning items would be like this image:

![Column Ordering Problem](https://user-images.githubusercontent.com/32509267/172038496-e72f675c-8cca-4d80-9b23-03e5494d2e22.jpg)

but that's not how you show items to people, it should be in Row Order, like this:

![Row Ordering](https://user-images.githubusercontent.com/32509267/172038018-be2bda7d-19fa-4d00-9b14-ef1d38de5511.jpg)

if I'm missing something, please let me know

**Comment by VladislavAntonyuk** [comment]
https://vladislavantonyuk.github.io/articles/.NET-MAUI-CollectionView-with-Staggered-Layout use as the start point. For iOS and Windows I will update a bit later
